x-pack/test/detection_engine_api_integration/security_and_spaces/group10/perform_bulk_action.ts

```diff
@@ -2997,5 +2997,223 @@ export default ({ getService }: FtrProviderContext): void => {
+          .expect('Content-Disposition', 'attachment; filename="rules_export.ndjson"')
+          .parse(binaryToString);
+
+        const [rule1, rule2, rule3, exportDetailsJson] = body.toString().split(/\n/);
+
+        const ruleWithLegacyInvestigationField = removeServerGeneratedProperties(JSON.parse(rule1));
+        expect(ruleWithLegacyInvestigationField.investigation_fields).to.eql({
+          field_names: ['client.address', 'agent.name'],
+        });
+
+        const ruleWithLegacyInvestigationFieldEmptyArray = removeServerGeneratedProperties(
+          JSON.parse(rule2)
+        );
+        expect(ruleWithLegacyInvestigationFieldEmptyArray.investigation_fields).to.eql(undefined);
+
+        const ruleWithInvestigationField = removeServerGeneratedProperties(JSON.parse(rule3));
+        expect(ruleWithInvestigationField.investigation_fields).to.eql({
+          field_names: ['host.name'],
+        });
+
+        const exportDetails = JSON.parse(exportDetailsJson);
+        expect(exportDetails).to.eql({
+          exported_exception_list_count: 0,
+          exported_exception_list_item_count: 0,
+          exported_count: 3,
+          exported_rules_count: 3,
+          missing_exception_list_item_count: 0,
+          missing_exception_list_items: [],
+          missing_exception_lists: [],
+          missing_exception_lists_count: 0,
+          missing_rules: [],
+          missing_rules_count: 0,
+          excluded_action_connection_count: 0,
+          excluded_action_connections: [],
+          exported_action_connector_count: 0,
+          missing_action_connection_count: 0,
+          missing_action_connections: [],
+        });
+      });
+
+      it('should delete rules', async () => {
+        const ruleIdWithLegacyInvestigationField = '2297be91-894c-4831-830f-b424a0ec84f0';
+        const ruleIdWithLegacyInvestigationFieldEmptyArray = '2297be91-894c-4831-830f-b424a0ec5678';
+        const ruleIdWithInvestigationField = '2297be91-894c-4831-830f-b424a0ec9102';
+
+        const { body } = await postBulkAction()
+          .send({ query: '', action: BulkActionType.delete })
+          .expect(200);
+
+        expect(body.attributes.summary).to.eql({ failed: 0, skipped: 0, succeeded: 3, total: 3 });
+
+        // Check that the deleted rule is returned with the response
+        const names = body.attributes.results.deleted.map(
+          (returnedRule: RuleResponse) => returnedRule.name
+        );
+        expect(names.includes('Test investigation fields')).to.eql(true);
+        expect(names.includes('Test investigation fields empty array')).to.eql(true);
+        expect(names.includes('Test investigation fields object')).to.eql(true);
+
+        // Check that the updates have been persisted
+        await fetchRule(ruleIdWithLegacyInvestigationField).expect(404);
+        await fetchRule(ruleIdWithLegacyInvestigationFieldEmptyArray).expect(404);
+        await fetchRule(ruleIdWithInvestigationField).expect(404);
```

Comment:
This test checks that the overall bulk deletion logic works, but doesn't check anything related to the investigation fields specifically. I think we should delete it or add any specific checks for the field.

Same for other tests:

- "should enable rules with legacy investigation fields"
- "should disable rules with legacy investigation fields"

Reply:
Updated tests to be more specific about what they are testing and if SO migration is inteded.